src/traits.d

```diff
@@ -134,11 +134,20 @@ extern (C++) Expression pointerBitmap(TraitsExp e)
         return new ErrorExp();
     }
 
-    d_uns64 sz = t.size(e.loc);
+    d_uns64 sz;
     if (t.ty == Tclass && !(cast(TypeClass)t).sym.isInterfaceDeclaration())
         sz = (cast(TypeClass)t).sym.AggregateDeclaration.size(e.loc);
+    else
+        sz = t.size(e.loc);
+
+    const sz_size_t = Type.tsize_t.size(e.loc);
+
+    if (sz == SIZE_INVALID || sz > sz.max - sz_size_t)
+    {
+        error(e.loc, "invalid size for type %s", t.toChars());
```

Comment:
Doesn't size failing indicate a forward reference error or an internal failure?  This is a user-facing error message, it would be better if it said something meaningful (or was turned into an ICE).